### PR TITLE
tags for the instructions.

### PR DESCRIPTION
**Pull request by Sanket-0510**
implements #5 the way described by issue #4.

I have tried covering tags for extension "C".  In last PR I was working on main branch hence overlooked extension function and some other check made by you. This PR will hopefully overcome all those mistakes and will do the checks for extension C.

### CODE REVIEW

File: model/riscv_insts_cext.sail [hunk 1]
```diff
@@ -426,201 +428,202 @@ function clause execute (C_ADDW(rsd, rs2)) = {
...
-      if rd != zreg & sizeof(xlen) == 64
+      if rd != zreg & sizeof(xlen) == 64 & extension("C")
 
 /* ****************************************************************** */
 union clause ast = C_SWSP : (bits(6), regidx)
 
-mapping clause encdec_compressed = C_SWSP(ui76 @ ui52, rs2)
-  <-> 0b110 @ ui52 : bits(4) @ ui76 : bits(2) @ rs2 : regidx @ 0b10
+mapping clause encdec_compressed = C_SWSP(ui76 @ ui52, rs2) if extension("C") if extension("C")
+  <-> 0b110 @ ui52 : bits(4) @ ui76 : bits(2) @ rs2 : regidx @ 0b10 if extension("C") if extension("C")
```
Comment (ThinkOpenly) [inline model/riscv_insts_cext.sail:527]: You have double adds here.

**Review by ThinkOpenly — CHANGES_REQUESTED** [review]
A few minor comments, but otherwise looks really close.
Have you attempted to test this by running with it? (That should've caught at least one of the issues.)

File: model/riscv_insts_cext.sail (hunk 1 above)
Comment (Sanket-0510) [reply model/riscv_insts_cext.sail:527]: Yes I overlooked those lines, I've fixed it. Let me know if any further changes required

**Comment by Sanket-0510** [comment]
> A few minor comments, but otherwise looks really close. Have you attempted to test this by running with it? (That should've caught at least one of the issues.)

All tests passed successfully on my side, I think we are good to go with this for extension C.
Screenshot from 2024-01-24 11-45-23

**Comment by ThinkOpenly** [comment]
@Sanket-0510, I reformatted your patches a bit, squashed them, committed into a fresh branch of my own, and merged them in PR #12. You are still listed as the author. I am just the committer. Thank you for your efforts here!

**Comment by ThinkOpenly** [comment]
FYI, this commit and other related "extension tagging" commits have been merged upstream. https://github.com/riscv/sail-riscv/commit/6492f8c728d2bf121521a5765f3e0613c08e5f2c. Congratulations, and thank you for your efforts! :tada:

**Pull request closed without merging**